code/atom/throwing.dm

```diff
@@ -25,6 +25,12 @@
 				if(!A.Cross(src))
 					src.throw_impact(A, thr)
 					. = TRUE
+			//Would be an idea to move all these checks into its own proc so non-humans don't need to check for this
+			if(ishuman(src) && istype(A, /obj/item/bamboo_spike_trap))
+				var/obj/item/bamboo_spike_trap/M = A
+				if(M.armed)
+					src.throw_impact(M, thr)
+					. = TRUE
```

Comment:
can't this just go in `bamboo_spike_trap/Crossed`? or is that not called for thrown things?

Reply:
It is theoretically called for thrown things iirc. The thing is i cannot stop the throw within crossed iirc. I made it like so so the person cannot be thrown into 3-4 traps with a single throw. If there is a proc that i can call to stop the throw withing bamboo_spike_trap/crossed, i will move it there of course.